packages/react-components/react-drawer/src/components/Drawer/useDrawerStyles.ts

```diff
@@ -1,33 +1,120 @@
+import { tokens } from '@fluentui/react-theme';
 
 export const drawerClassNames: SlotClassNames<DrawerSlots> = {
   root: 'fui-Drawer',
-  // TODO: add class names for all slots on DrawerSlots.
-  // Should be of the form `<slotName>: 'fui-Drawer__<slotName>`
+  dialog: 'fui-Drawer__dialog',
+  dialogSurface: 'fui-Drawer__dialogSurface',
+};
+
+/**
+ * CSS variable names used internally for uniform styling in Drawer.
+ */
+export const drawerCSSVars = {
+  size: '--fui-Drawer--size',
 };
 
 /**
  * Styles for the root slot
  */
 const useStyles = makeStyles({
   root: {
-    // TODO Add default styles for the root element
+    ...shorthands.padding('16px'),
+    ...shorthands.borderRadius(0),
+    ...shorthands.border(0),
+
+    boxSizing: 'border-box',
+    width: `var(${drawerCSSVars.size})`,
+    maxWidth: 'calc(100vw - 48px)',
+    height: 'auto',
+    top: 0,
+    bottom: 0,
+  },
+
+  leftDrawer: {
+    left: 0,
+    right: 'auto',
   },
 
-  // TODO add additional classes for different states and/or slots
+  rightDrawer: {
+    right: 0,
+    left: 'auto',
+  },
+
+  inline: {
+    position: 'relative',
+    alignItems: 'stretch',
+    justifyContent: 'stretch',
+  },
+
+  inlineLeft: {
+    ...shorthands.borderRight('1px', 'solid', tokens.colorNeutralBackground3),
+  },
+
+  inlineRight: {
+    ...shorthands.borderLeft('1px', 'solid', tokens.colorNeutralBackground3),
+  },
+
+  temporary: {
+    position: 'fixed',
+  },
+
+  sizeSmall: {
+    [drawerCSSVars.size]: '320px',
+  },
+  sizeMedium: {
+    [drawerCSSVars.size]: '592px',
+  },
+  sizeLarge: {
+    [drawerCSSVars.size]: '940px',
+  },
+  sizeFull: {
+    [drawerCSSVars.size]: '100vw',
+    maxWidth: '100vw',
+  },
 });
 
+function getInlineClasses(state: DrawerState, styles: ReturnType<typeof useStyles>) {
+  const classes = [styles.inline];
+
+  if (state.position === 'left') {
+    classes.push(styles.inlineLeft);
+  }
+
+  if (state.position === 'right') {
+    classes.push(styles.inlineRight);
+  }
+
+  return mergeClasses(...classes);
+}
+
+const sizeMap = {
+  small: 'sizeSmall',
+  medium: 'sizeMedium',
+  large: 'sizeLarge',
+  full: 'sizeFull',
+} as const;
+
 /**
  * Apply styling to the Drawer slots based on the state
  */
 export const useDrawerStyles_unstable = (state: DrawerState): DrawerState => {
   const styles = useStyles();
-  state.root.className = mergeClasses(drawerClassNames.root, styles.root, state.root.className);
 
-  // TODO Add class names to slots, for example:
-  // state.mySlot.className = mergeClasses(styles.mySlot, state.mySlot.className);
+  const baseClasses = [
+    drawerClassNames.root,
+    styles.root,
+    state.position === 'left' ? styles.leftDrawer : styles.rightDrawer,
+    typeof state.size !== 'number' && styles[sizeMap[state.size]],
+  ];
+
+  if (state.type === 'inline') {
+    state.root.className = mergeClasses(...baseClasses, getInlineClasses(state, styles), state.root.className);
+  } else if (state.dialogSurface) {
```

Comment:
using `else if` is usually an code-smell. It confuses the intent as it handles 2 separate scenarios from my POV, but I might be wrong. anyways can we remove it and use early exit ?

Reply:
We can't use an early exit, as we mutate the state and then return it. If we do this, then it means we would need to have the same return, which is very confusing as well:
![image](https://user-images.githubusercontent.com/1357885/230088899-5b02fd17-6962-4eb3-b7d4-7056598917a5.png)

Alternatively, we could go to something like this, but it'll be a double check':
![image](https://user-images.githubusercontent.com/1357885/230089265-3ebc6f2c-d79b-4b94-a7f3-aa89d866a2d1.png)